app/profile/test/task.form.modal/task.form.modal.controller.js

```diff
@@ -6,11 +6,37 @@
         .module('app')
         .controller('TaskFormModalController', TaskFormModalController);
 
-    function TaskFormModalController ($uibModalInstance) {
+    function TaskFormModalController (currentTest, $uibModalInstance, $http, Endpoints, growl, $log) {
         const vm = this;
 
+        vm.submit = submit;
         vm.close = close;
 
+        ////
+
+        function submit () {
+
+            if (!vm.form || vm.form.$invalid) {
+                return vm.form.$submitted = false;
+            }
+
+            let data = {
+                name: vm.task.name,
+                description: vm.task.description,
+                testId: currentTest.id
+            };
+
+            $http.post(Endpoints.ACHIEVEMENTS_TASK, data).then((res) => {
```

Comment:
The better way to make the request is to use angular factory instead of just $http service. That's ok but try to create a factory for ACHIEVEMENTS_TASK. As example you can use file ./app/utils/factories/user.factory.js